src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const arr = [];
+  let copyOfState = { ...state };
+
+  for (const values of actions) {
+    if (values.type === 'addProperties') {
+      Object.assign(copyOfState, values.extraData);
+      arr.push({ ...copyOfState });
+    }
+
+    if (values.type === 'removeProperties') {
+      for (const keysToRemoveValues of values.keysToRemove) {
```

Comment:
Would be better to use something like so
```
for (const key of values.keysToRemove) {
  delete copyOfState[key];
}
```
So in this case we don't need to compare we just delete all of the required keys